Symfony/CS/Fixer/Contrib/EregToPregFixer.php

```diff
@@ -0,0 +1,171 @@
+        array('eregi', 'preg_match', 'i'),
+        array('ereg_replace', 'preg_replace', ''),
+        array('eregi_replace', 'preg_replace', 'i'),
+        array('split', 'preg_split', ''),
+        array('spliti', 'preg_split', 'i'),
+    );
+
+    /**
+     * @var array
+     */
+    private $delimiters = array('/', '#', '!');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        if (!$this->cursoryMatch($content)) {
+            return $content;
+        }
+
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($this->map as $map) {
+            $seq = array(array(T_STRING, $map[0]), '(');
+            $currIndex = 0;
+            while (null !== ($ereg = $tokens->findSequence($seq, $currIndex, $tokens->count(), false))) {
+                // Ensure it's a function call
+                $prev = $tokens->getPrevMeaningfulToken($ereg[0]);
+                if (!isset($prev) || $tokens[$prev]->isGivenKind(array(T_OBJECT_OPERATOR, T_DOUBLE_COLON))) {
+                    break;
+                }
+
+                // Extract the string to convert
+                $index = $tokens->getNextMeaningfulToken($ereg[1]);
+                if (!isset($index) || !$tokens[$index]->isGivenKind(T_CONSTANT_ENCAPSED_STRING)) {
+                    break;
+                }
+
+                // Ensure the string is not appended something or worse
+                $next = $tokens->getNextMeaningfulToken($index);
+                if (!isset($next) || !in_array($tokens[$next]->getContent(), array(',', ')'))) {
+                    break;
+                }
+
+                // Advance tokenizer cursor
+                $currIndex = $index;
+
+                // Convert to PCRE
+                $string = substr($tokens[$index]->getContent(), 1, -1);
+                $quote = substr($tokens[$index]->getContent(), 0, 1);
+                $delim = $this->getBestDelimiter($string);
+                $preg = $delim.addcslashes($string, $delim).$delim.'D'.$map[2];
+
+                // Ensure  the preg is valid
+                if (!$this->checkPreg($preg)) {
+                    continue;
+                }
+
+                // Modify function and argument
+                $tokens[$index]->setContent($quote.$preg.$quote);
+                $tokens[$ereg[0]]->setContent($map[1]);
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Replace deprecated ereg regular expression functions with preg.';
+    }
+
+    /**
+     * Check the validity of a PCRE
+     *
+     * @param $preg the regular expression
+     *
+     * @return bool
+     */
+    private function checkPreg($preg)
+    {
+        $ok = true;
+
+        set_error_handler(function () use (&$ok) {
+            $ok = false;
+        }, E_WARNING);
+
+        preg_match($preg, '');
+
+        restore_error_handler();
```

Comment:
rather than using a custom error handler for that and catching the warning, I would rather rely on the ``false`` return value of ``preg_match`` (silencing the warning)